src/routes/accommodationRoute.js

```diff
@@ -83,21 +83,23 @@ const fUpload = fileUpload({
 });
 
 const {
-  isSupplierAccommodation, checkForImages, checkForImagesUpdate
+  isSupplierAccommodation, checkForImages, checkForImagesUpdate, accommodationExists
 } = AccommodationMiddleware;
 const { checkToken } = UserMiddleware;
 
 app.patch('/:id', fUpload, checkToken, RoleMiddleware.checkHostOrTAdmin, isSupplierAccommodation, checkForImagesUpdate, valid.editAccommodation, AccommodationsController.editAccommodation);
 app.delete('/:id', checkToken, RoleMiddleware.checkHost, isSupplierAccommodation, AccommodationsController.deleteAccommodation);
 app.post('/', fUpload, checkToken, RoleMiddleware.checkHostOrTAdmin, valid.accommodation, checkForImages, AccommodationsController.addAccommodation);
 app.get('/', AccommodationsController.getAccommodations);
-app.get('/:accommodationId/rating', valid.getReviewvalidation, BookingsController.getRating);
+app.get('/:accommodationId/rating', valid.accommodationId, BookingsController.getRating);
 app.get('/search', AccommodationsController.getAccommodationsByFilter);
+app.get('/bookmarked', checkToken, AccommodationsController.getBookmarkedAccommodations);
 app.get('/:accommodationId', AccommodationsController.getAccommodationById);
 app.post('/:accommodationId/like', [
   checkToken,
   LikingsController.addLikeAccommdation
 ]);
-
-
+app.post('/:accommodationId/bookmark', [checkToken, valid.accommodationId,
+  accommodationExists,
+  AccommodationsController.bookmarkAccommodation]);
```

Comment:
nice job so far.
However, as the story says `enable bookmarking` which also includes `un-bookmarking` I suggest you separate the functionalities for bookmarking and un-bookmarking into different routes.
it won't take many functionalities.

Reply:
great!